test/unit/SmartContracts/X509.unit.test.mjs

```diff
@@ -107,16 +161,34 @@ describe('DerParser contract functions', function () {
       certChain[1].tlvLength,
       0,
       false,
+      0,
     );
-    // now presenting the end user cert should work because the smart contract now trusts the Intermediate CA public key
+    // now presenting the end user cert should also work
     await X509Instance.validateCertificate(
       certChain[0].derBuffer,
       certChain[0].tlvLength,
       signature,
       true,
+      0,
```

Comment:
Hi @Westlad, this the index of the `oidGroup` right? I didn't like much this because the user must know the index. Wouldn't we be able to use the `OID Group` value as the index? (e.g. `2.16.840.1.114412.3.21.2`)

Reply:
Ah, no, that's not what the `oidGroup` is.  The example you gave is a single OID.  However a set (array) of OIDs must be applied to the certificate check.  This set is dependent on the CA being used. So for example Digicert might be `oidGroup = 1`, Entrust is `oidGroup = 2` etc. . Each index choice gives a different array of OIDs.  It's true that the user will need to know which `oidGroup` index to use with their certificate but we expect the person deploying the contracts to publish a mapping of `CA => oidGroup`.

Reply:
You're right, I made confusion.